oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -667,14 +691,8 @@ String createAssertion(JsonFactory jsonFactory, long currentTime, String audienc
       payload.setAudience(audience);
     }
 
-    String assertion;
-    try {
-      assertion = JsonWebSignature.signUsingRsaSha256(privateKey, jsonFactory, header, payload);
```

Comment:
Critical: I'm not sure the new code is using this algorithm. I might need to patch this into an IDE and check.

Reply:
The signing algorithm is dictated by `OAuth2Utils.SIGNATURE_ALGORITHM`. Which is "SHA256withRSA"

Reply:
OK. Since this is security critical code I'm going to need go over this carefully and make sure of that. It might take a little time to review, or perhaps someone else can get to this before I do.